lndp_apply_perts.F90

```diff
@@ -35,41 +40,82 @@ subroutine lndp_apply_perts(blksz,lsm, lsoil,dtf, n_var_lndp, lndp_var_list, &
         real(kind=kind_dbl_prec),     intent(in) :: stype(:,:)
+        real(kind=kind_dbl_prec),     intent(in) :: maxsmcnoah(:) 
         real(kind=kind_dbl_prec),     intent(in) :: maxsmc(:) 
+        real(kind=kind_dbl_prec),     intent(in) :: zs_lsm(:)
 
         ! intent(inout) 
         real(kind=kind_dbl_prec),     intent(inout) :: smc(:,:,:)
         real(kind=kind_dbl_prec),     intent(inout) :: slc(:,:,:)
         real(kind=kind_dbl_prec),     intent(inout) :: stc(:,:,:)
         real(kind=kind_dbl_prec),     intent(inout) :: vfrac(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: snoalb(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: alvsf(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: alnsf(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: alvwf(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: alnwf(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: facsf(:,:)
+        real(kind=kind_dbl_prec),     intent(inout) :: facwf(:,:)
+        !real(kind=kind_dbl_prec),     intent(inout) :: semis(:,:)
 
         ! intent(out) 
         integer,                        intent(out) :: ierr
 
         ! local
         integer         :: nblks, print_i, print_nb, i, nb
-        integer         ::  this_im, v, soiltyp, k
+        integer         :: this_im, v, soiltyp, k, nsoil
         logical         :: print_flag 
 
         real(kind=kind_dbl_prec) :: p, min_bound, max_bound, tmp_sic,  pert
+        real(kind=kind_dbl_prec), dimension(max(lsoil,lsoil_lsm)) :: zslayer, smc_vertscale, stc_vertscale
 
         ! decrease in applied pert with depth
-        real(kind=kind_dbl_prec), dimension(4), parameter  :: smc_vertscale = (/1.0,0.5,0.25,0.125/)
-        real(kind=kind_dbl_prec), dimension(4), parameter  :: stc_vertscale = (/1.0,0.5,0.25,0.125/)
-
-        ! model-dependent values, hard-wired in noah code.
+        !-- Noah lsm
+        real(kind=kind_dbl_prec), dimension(4), parameter  :: smc_vertscale_noah = (/1.0,0.5,0.25,0.125/)
+        real(kind=kind_dbl_prec), dimension(4), parameter  :: stc_vertscale_noah = (/1.0,0.5,0.25,0.125/)
         real(kind=kind_dbl_prec), dimension(4), parameter  :: zs_noah = (/0.1, 0.3, 0.6, 1.0/)
-        real(kind=kind_dbl_prec), parameter                :: minsmc = 0.02
+        !-- RUC lsm
+        real(kind=kind_dbl_prec), dimension(9), parameter :: smc_vertscale_ruc = (/1.0,0.9,0.8,0.6,0.4,0.2,0.1,0.05,0./)
+        real(kind=kind_dbl_prec), dimension(9), parameter :: stc_vertscale_ruc = (/1.0,0.9,0.8,0.6,0.4,0.2,0.1,0.05,0./)
+        !
+        real(kind=kind_dbl_prec), parameter               :: minsmc = 0.02
+        real(kind=kind_dbl_prec), parameter               :: zsbot  = 6.
 
         ierr = 0 
 
-        if (lsm .NE. 1 ) then 
-                write(6,*) 'ERROR: lndp_apply_pert assumes LSM is noah, ', & 
+        if (lsm .NE. 1 .and. lsm .ne. lsm_ruc) then 
```

Comment:
`.NE. 1` should become `.ne. lsm_noah` (and `lsm_noah` should be passed in via the argument list)